api/grpcserver/mesh_service.go

```diff
@@ -111,7 +114,140 @@ func (s MeshService) MaxTransactionsPerSecond(ctx context.Context, in *pb.MaxTra
 // AccountMeshDataQuery returns account data
 func (s MeshService) AccountMeshDataQuery(ctx context.Context, in *pb.AccountMeshDataQueryRequest) (*pb.AccountMeshDataQueryResponse, error) {
 	log.Info("GRPC MeshService.AccountMeshDataQuery")
-	return nil, nil
+
+	// This is the latest layer we're aware of
+	//latestLayer := s.Mesh.LatestLayer()
+	startLayer := types.LayerID(in.MinLayer)
+
+	if startLayer > s.Mesh.LatestLayer() {
+		return nil, status.Errorf(codes.InvalidArgument, "`LatestLayer` must be less than or equal to latest layer")
+	}
+	if in.Filter == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "`Filter` must be provided")
+	}
+	if in.Filter.AccountId == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "`Filter.AccountId` must be provided")
+	}
+
+	// Gather transaction data
+	//txs := &pb.AccountMeshData{DataItem: &pb.AccountMeshData_Transaction{Transaction: }
+	addr := types.BytesToAddress(in.Filter.AccountId.Address)
+	res := &pb.AccountMeshDataQueryResponse{}
+	if in.Filter.AccountMeshDataFlags&uint32(pb.AccountMeshDataFlag_ACCOUNT_MESH_DATA_FLAG_TRANSACTIONS) != 0 {
```

Comment:
why not have different endpoints for different data types?

Reply:
The *Query and *Stream endpoints form a pair. In this case, as discussed on the design review for https://github.com/spacemeshos/SMIPS/issues/13 a few weeks ago, we decided to multiplex multiple, related data structures, such as transactions and activations, onto a single stream with a filter (see also https://github.com/spacemeshos/api/issues/40). If we're going to have multiplexed streams, why not have a query that does basically the same thing, and accepts the same request type? I think this makes it a little easier for a client that wants to sync everything - they first hit a *Query endpoint to get caught up, then switch to the *Stream endpoint to stay current.

CC @avive

Reply:
following our conversation, I think that different endpoints for query may be easier to maintain and understand. if not, I'd recommend using options go pattern to implement different filters

Reply:
@antonlerner Thanks for sharing the functional options pattern - I actually used this in another part of the API implementation later on. But I'm a little confused how it could be used for query filters. How would this be different than or better than the existing filter design? (see e.g.: https://github.com/spacemeshos/api/blob/ff4a9a6b913a23a83cf07a0a7b4b7c16605ce9c0/proto/spacemesh/v1/mesh_types.proto#L56)